tasks/main.yml

```diff
@@ -61,6 +61,25 @@
   when: firewalld_purge_services and item not in firewalld_service_rules
   tags: firewalld
 
+- name: get active firewalld port rules
+  shell: set -o pipefail; /bin/firewall-cmd --list-ports | sed -e 's/ /\n/g'
+  args:
+    executable: /bin/bash
+  register: active_ports
+  changed_when: false
+  tags: firewalld
+
+- name: purge unconfigured firewalld port rules
+  firewalld:
+    port: "{{ item }}"
+    permanent: true
+    immediate: true
+    state: disabled
+    zone: "{{ default_zone|default('public') }}"
+  with_items: "{{ active_ports.stdout_lines }}"
+  when: firewalld_purge_services and item not in firewalld_service_rules
```

Comment:
The conditional should be `when: firewalld_purge_ports and item not in firewalld_port_rules`. This way the purge of unconfigured ports can be configured separately from services. Also the conditional have to check if its item is in `firewalld_port_rules` to avoid removal and recreation of configured ports. I'll fix it after merging your code.